src/Products/PythonScripts/PythonScript.py

```diff
@@ -195,6 +195,7 @@ def ZPythonScriptHTML_upload(self, REQUEST, file=''):
                     manage_tabs_message='No file specified',
                     manage_tabs_type='warning')
             file = file.read()
+            file = file.decode('utf-8')
```

Comment:
Do the decoding in `write` - and **test** the string to see if it is bytes, first.